### PR TITLE
[lambda] Modify how `DD_APM_FLUSH_DEADLINE_MILLISECONDS` is handled

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Update the environment variable `DD_APM_FLUSH_DEADLINE` to `DD_APM_FLUSH_DEADLINE_MILLISECONDS`.
Updated how `DD_APM_FLUSH_DEADLINE_MILLISECONDS` is handled. 
Previously, we would set the deadline as the environment variable value, if set. Now, we take the deadline as the remaining time in an AWS Lambda, and the value in the environment variable is used to subtract from the deadline. The default is now 100ms.

This is not a breaking change, since the release was not GA for Lambda. 

### Motivation

Updated how `DD_APM_FLUSH_DEADLINE_MILLISECONDS` is handled. 
Have units on the environment variable. 
Updating to match the environment variable in `dd-trace-py`. Doing it before we have GA on our instrumentation tools for AWS Lambda.

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

**Review by Qard — APPROVED** [review]

**Review by astuyve — DISMISSED** [review]

File: packages/dd-trace/src/lambda/handler.js [hunk 1]
```diff
@@ -23,7 +23,13 @@ let __lambdaTimeout
  */
 function checkTimeout (context) {
   let remainingTimeInMillis = context.getRemainingTimeInMillis()
-  const apmFlushDeadline = parseInt(process.env.DD_APM_FLUSH_DEADLINE)
+
+  let apmFlushDeadline = process.env.DD_APM_FLUSH_DEADLINE_MILLISECONDS
+  if (!apmFlushDeadline) {
+    apmFlushDeadline = process.env.DD_APM_FLUSH_DEADLINE
```
Comment (astuyve) [inline packages/dd-trace/src/lambda/handler.js:29]: Maybe add a comment to remove this block with the next major? Or even in v3?

**Review by astuyve — APPROVED** [review]

File: packages/dd-trace/src/lambda/handler.js [hunk 2]
```diff
@@ -23,7 +23,15 @@ let __lambdaTimeout
  */
 function checkTimeout (context) {
   let remainingTimeInMillis = context.getRemainingTimeInMillis()
-  const apmFlushDeadline = parseInt(process.env.DD_APM_FLUSH_DEADLINE)
+
+  let apmFlushDeadline = process.env.DD_APM_FLUSH_DEADLINE_MILLISECONDS
+  // TODO: Remove this condition and parse int above
+  // once `datadog-lambda-js` updates tracer to v3
+  if (!apmFlushDeadline) {
```
Comment (astuyve) [inline packages/dd-trace/src/lambda/handler.js:30]: Update - we learned that the python change was not released yet so this env var can be removed w/o being backwards incompatible. Going to remove this conditional and tweak the logic a bit.

Comment (duncanista) [reply packages/dd-trace/src/lambda/handler.js:28]: updated logic and name

**Review by astuyve — APPROVED** [review]
Thanks for the update here!

File: packages/dd-trace/src/lambda/handler.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ function checkTimeout (context) {
   const remainingTimeInMillis = context.getRemainingTimeInMillis()
 
   let apmFlushDeadline = parseInt(process.env.DD_APM_FLUSH_DEADLINE_MILLISECONDS)
-  if (!apmFlushDeadline) {
+  if (!apmFlushDeadline || apmFlushDeadline < 0) {
```
Comment (astuyve) [inline packages/dd-trace/src/lambda/handler.js:28]: Let's keep this consistent between node and [python](https://github.com/DataDog/dd-trace-py/pull/5176/files#diff-6c80e8ef43534becbd3732c7b7631ac8a28b559325e8286df77e2e7dd7e6e9e1R66). `if apmFlushDeadline < 0`, just use `0`

Comment (duncanista) [reply packages/dd-trace/src/lambda/handler.js]: Updated the one in python to default to a `100` too, as discussed in our meeting

**Review by astuyve — APPROVED** [review]

**Pull request merged**